lib/Dialect/FIRRTL/Transforms/InferWidths.cpp

```diff
@@ -1179,11 +1230,30 @@ bool InferenceTypeUpdate::updateValue(Value value) {
 
   // Get the inferred width.
   Expr *expr = mapping.getExprOrNull(value);
-  if (!expr || !expr->solution.hasValue())
+  if (!expr || !expr->solution.hasValue()) {
+    anyFailed = true;
+    auto op = value.getDefiningOp();
+    // Emit an error that indicates where we have failed to infer a width. Note
+    // that we only get here if the IR contained some operation or FIRRTL type
+    // that is not yet supported by width inference. Errors related to widths
+    // not being inferrable due to contradictory constraints are handled earlier
+    // in the solver, and the pass never proceeds to perform this type update.
+    // TL;DR: This is for compiler hackers.
+    // TODO: Convert this to an assertion once we support all operations and
+    // types for width inference.
+    auto diag = mlir::emitError(op ? op->getLoc() : value.getLoc(),
```

Comment:
I don't understand the loc gymnastics, can't you just use `value.getLoc()` without the ternary?

Reply:
Yes totally, that's just me being a derp.